adyenv6b2ccheckoutaddon/acceleratoraddon/web/src/com/adyen/v6/utils/SameSiteCookieAttributeAppenderUtils.java

```diff
@@ -0,0 +1,134 @@
+/*
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Hybris Extension
+ *
+ * Copyright (c) 2020 Adyen B.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ */
+
+package com.adyen.v6.utils;
+
+import com.google.common.net.HttpHeaders;
+import de.hybris.platform.servicelayer.config.ConfigurationService;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.configuration.Configuration;
+import org.apache.log4j.Logger;
+
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/*
+ * This class uses code written by Igor Zarvanskyi and published on https://clutcher.github.io/post/hybris/same_site_login_issue/
+ */
+public class SameSiteCookieAttributeAppenderUtils {
+
+    private static final Logger LOG = Logger.getLogger(SameSiteCookieAttributeAppenderUtils.class);
+
+    private ConfigurationService configurationService;
+
+    private static final String PLATFORM_VERSION_PROPERTY = "build.version.api";
+    private static final String SAMESITE_COOKIE_HANDLER_ENABLED_PROPERTY = "adyen.samesitecookie.handler.enabled";
+    private static final List<String> SAP_VERSIONS_WITH_SAMESITE_FIX = Collections.singletonList("2005");
```

Comment:
we can add check for less than 2005 because when 21XX will be released, this won't work.

Reply:
I tried converting the string version value we retrieve from the platform to a numeric value and check it as `platformVersion <= 2005`. But the platform value can be something like "6.7.0" which is not convertible to a numeric.

One idea I had was normalising it removing the dots, i.e. "670". 
I couldn't find a list of all possible version values, so if something like "x.x.10" exists, it could result in a number higher than 2005 when it's actually an older version. But we can go with such approach if no such version exists.

As for the solution we currently have on the code, it would need a manual update of the list before we officially confirm the plugin as compatible with any version released after 2005.

Reply:
lets check if (version==2005||regular_expression_for_version matches 6.X).

Reply:
fixed, now getting the major version from the string and then comparing to 2005